docs: add delete groups

sarahxsanders opened the pull request:
## Changes
- Removed limitation on deleting groups, added new section for deleting groups, fix for https://github.com/PostHog/posthog/issues/10953#issuecomment-3548596804
- Removed line breaks as they were creating some odd spacing in the bulleted list
- Updated all screenshots w/ light and dark mode
- Renamed some headings - I prefer action oriented headings over "how to" or gerunds, "how to" is good for the title of a tutorial, but can impact search/LLM crawling
- Updated outdated UI components
- Added another relevant further reading resources 

## Preview
https://posthog-git-delete-group-types-post-hog.vercel.app/docs/product-analytics/group-analytics

## Checklist

- [x] Words are spelled using American English
- [x] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [x] Feature names are in **[sentence case too](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)**. It's "product analytics" not "Product Analytics" and so on.
- [x] Use relative URLs for internal links

gewenyu99 commented on contents/docs/product-analytics/group-analytics.mdx:
```diff
@@ -13,25 +13,29 @@ import JSCreateGroups from "../_snippets/how-to-create-groups.mdx"
 export const groupsInFeatureDark = "https://res.cloudinary.com/dmukukwp6/video/upload/posthog.com/contents/images/docs/user-guides/group-analytics/groups-in-feature-flags-dark-mode.mp4"
 export const groupsProjectSettingLight = "https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/docs/user-guides/group-analytics/groups-project-settings-light-mode.png"
 export const groupsProjectSettingDark = "https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/docs/user-guides/group-analytics/groups-project-settings-dark-mode.png"
+export const deleteGroupsLight = "https://res.cloudinary.com/dmukukwp6/image/upload/q_auto,f_auto/Screenshot_2025_11_18_at_5_24_35_PM_dd4232b4cd.png"
```
I think we can make these narrower and more focused, this is really wide :P

gewenyu99 commented on contents/docs/product-analytics/group-analytics.mdx:
```diff
@@ -208,27 +206,51 @@ if (isFlagEnabled) {
 
+## Delete groups
+
+To delete a group type:
```
```suggestion
To delete a group:
```

arthurdedeus replied:
Not sure about this. The section above is named `Rename group types` and this one is called `Delete groups`.

IMO we should be very clear when we mean group as an individual group or a group type, as deleting the former is not possible, while the latter is. Using "delete a group" seems to me that we're deleting an individual group, not a category of groups (which is the case).

Not sure if `group type` is the best customer-facing term here either, even though is the one we use in the code.

gewenyu99 reviewed (approved):
Noice, some comments

luke-belton commented on contents/docs/product-analytics/group-analytics.mdx:
```diff
@@ -208,27 +206,51 @@ if (isFlagEnabled) {
+
+<CalloutBox icon="IconWarning" title="Important notes about deleting groups" type="caution">
+
+- When you delete a group, the underlying event data remains unchanged. The group data is filtered out from queries based on a deletion timestamp, which means deleted groups do not count toward your usage limits.
```
> which means deleted groups do not count toward your usage limits.

is this true? I'm not super familiar with the pricing for groups. Obviously we'll have already charged for the events received up to deletion, and I thought because groups reappear when we receive new events we'll immediately start charging again?

arthurdedeus replied:
Usage limits here sounds ambiguous. I think the point is that the deleted group types won't count towards the 5 group types hard limit that we have.

arthurdedeus commented on contents/docs/product-analytics/group-analytics.mdx:
```diff
@@ -208,27 +206,51 @@ if (isFlagEnabled) {
+- When you delete a group, the underlying event data remains unchanged. The group data is filtered out from queries based on a deletion timestamp, which means deleted groups do not count toward your usage limits.
+- Historical events associated with the deleted group are preserved in the database. If new events come in with the same group key, the group will reappear in the interface.
+- The group key can be reused after deletion to create a new group, and any new events with that group key will cause it to show up again.
+- Deleting a group is permanent and cannot be undone through the interface.
```
Let's say I have a group type called "Orgs", which has events. If I delete "Orgs", all existing events will not be accessible when filtering by "Orgs" anymore.

If we ingest a new event with "Orgs" as group type, the group type will be recreated with a newer `created_at` timestamp. This means that the old events (with timestamp older than `created_at`) won't show up for this group type, even though it has the same name as before.

sarahxsanders commented:
@gewenyu99 @luke-belton @arthurdedeus thank you for all the feedback! I went ahead and decided to do a big overhaul on this page, rather than just a smaller update

the most recent commit addresses all your feedback, plus: 
- this one is more for @gewenyu99 - I consolidated all the content from snippets into one file. I honestly find it really disruptive to update content in snippets unless there is a good reason for keeping content in a snippet (like reuse). I didn't find we were reusing any of them elsewhere. We can discuss as a docs team and maybe set a standard for when we use them, because editing 7 files to update one page slows me down personally.
- Fixed group type vs individual group terminology confusion (see key concepts and new section headings)

arthurdedeus commented on contents/docs/product-analytics/group-analytics.mdx:
```diff
@@ -7,59 +7,548 @@ availability:
+<CalloutBox icon="IconInfo" title="New: B2B analytics preview" type="fyi">
 
-<JSGroupsIntro />
+Group analytics is getting a makeover! Sign up for the [B2B analytics feature preview](https://app.posthog.com/settings/user-feature-previews#b2b-analytics) to get a sneak peek, and [share your feedback](https://app.posthog.com/#panel=support%3Afeedback%3Acrm%3Alow).
```
```suggestion
Group analytics is getting a makeover! Sign up for the [customer analytics feature preview](https://app.posthog.com/settings/user-feature-previews#customer-analytics-roadmap) to get a sneak peek, and [share your feedback](https://app.posthog.com/#panel=support%3Afeedback%3Acrm%3Alow).
```
